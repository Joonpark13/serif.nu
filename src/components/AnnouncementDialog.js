import React, { useState } from 'react';
import { makeStyles } from '@material-ui/styles';
import { Dialog, DialogActions, DialogTitle, DialogContent, DialogContentText, Button } from '@material-ui/core';

export const useStyles = makeStyles({
  dialog: {
    top: '12.5%',
    left: '12.5%',
    width: '500px',
    padding: '10px',
  },
});

export default function AnnouncementDialog() {
  const classes = useStyles();
  const [showDialog, setShowDialog] = useState(!window.localStorage.getItem('announcementSeen'));

  const closeDialog = () => {
    window.localStorage.setItem('announcementSeen', true);
    setShowDialog(false);
  };

  return (
    <Dialog
      open={showDialog}
      onClose={closeDialog}
    >
      <div className={classes.dialog}>
        <DialogTitle variant="h5">
          Future plans for Serif.nu
        </DialogTitle>
        <DialogContent>
          <DialogContentText>
            It has been quite the ride for Serif over the past few years. What initially started as
            a small personal project by Joon has become an integral part of the class registration
            experience, having now helped Northwestern students for almost
            five years.
          </DialogContentText>
          <DialogContentText>
            Unfortunately, the&nbsp;
            <a href="/about" target="_blank" rel="noopener noreferrer">dev team</a>
            &nbsp;will be dissolving as Kevin graduates this upcoming June.
            This means that active development and maintenance of Serif.nu will cease unless a
            capable and dedicated developer takes over ownership of the project. Maintenance entails
            paying for the domain, hosting and deploying the source code for Serif, and updating the
            classes every quarter.
          </DialogContentText>
          <DialogContentText>
            You can contact Kevin at kevinlee2020@u.northwestern.edu if you are interested in
            doing so. Serif.nu is open-source, the code is available to see on&nbsp;
            <a href="https://github.com/Joonpark13/serif.nu" target="_blank" rel="noopener noreferrer">GitHub.</a>
          </DialogContentText>
          <DialogContentText>
            Thanks for using Serif.nu!
          </DialogContentText>
          <DialogContentText>
            Kevin, Joon, and the Serif.nu dev team
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={closeDialog}>Close</Button>
        </DialogActions>
      </div>
    </Dialog>
  );
}
