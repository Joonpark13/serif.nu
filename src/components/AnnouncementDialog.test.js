import React from 'react';
import { shallow } from 'enzyme';
import AnnouncementDialog from './AnnouncementDialog';

describe('AnnouncementDialog', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<AnnouncementDialog />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
