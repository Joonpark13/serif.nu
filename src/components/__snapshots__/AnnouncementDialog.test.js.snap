// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnouncementDialog should render correctly 1`] = `
<ForwardRef(WithStyles)
  onClose={[Function]}
  open={true}
>
  <div
    className="makeStyles-dialog-1"
  >
    <ForwardRef(WithStyles)
      variant="h5"
    >
      Future plans for Serif.nu
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        It has been quite the ride for Serif over the past few years. What initially started as a small personal project by Joon has become an integral part of the class registration experience, having now helped Northwestern students for almost five years.
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        Unfortunately, the 
        <a
          href="/about"
          rel="noopener noreferrer"
          target="_blank"
        >
          dev team
        </a>
         will be dissolving as Kevin graduates this upcoming June. This means that active development and maintenance of Serif.nu will cease unless a capable and dedicated developer takes over ownership of the project. Maintenance entails paying for the domain, hosting and deploying the source code for Serif, and updating the classes every quarter.
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        You can contact Kevin at kevinlee2020@u.northwestern.edu if you are interested in doing so. Serif.nu is open-source, the code is available to see on 
        <a
          href="https://github.com/Joonpark13/serif.nu"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub.
        </a>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        Thanks for using Serif.nu!
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        Kevin, Joon, and the Serif.nu dev team
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        onClick={[Function]}
      >
        Close
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </div>
</ForwardRef(WithStyles)>
`;
